packages/docgen/lib/get-type-annotation.js

```diff
@@ -13,7 +13,7 @@ const { types: babelTypes } = require( '@babel/core' );
 
 /**
  * @param {babelTypes.TSCallSignatureDeclaration | babelTypes.TSFunctionType | babelTypes.TSConstructSignatureDeclaration} typeAnnotation
- * @param {' => ' | ': '} returnIndicator The return indicator to use. Allows using the same function for function annotations and object call properties.
+ * @param {' => ' | ': '}                                                                                                  returnIndicator The return indicator to use. Allows using the same function for function annotations and object call properties.
```

Comment:
Not the best readability 😅 

Perhaps the `typeAnnotation` type could be moved to a `@typedef`? That would help improve alignment.

Reply:
Sure! Makes sense! I updated it here: https://github.com/WordPress/gutenberg/pull/25300/commits/56b0a2c1f1504aeba0622e041c8890200c2d7884

Since we already had a `TypeAnnotation` typedef, I named it `ExtendedTypeAnnotation`. Do you think it makes sense here?

I think we have a few other similar cases in the code that would demand manual improvements, but I tried not to fix all of them, since they also need some context understanding.